chainer/datasets/mnist.py

```diff
@@ -0,0 +1,107 @@
+import gzip
+import os
+import struct
+
+import numpy
+import six
+
+from chainer.dataset import download
+from chainer.datasets import tuple_dataset
+
+
+def get_mnist(withlabel=True, ndim=1, scale=1.):
+    """Gets the MNIST dataset.
+
+    `MNIST <http://yann.lecun.com/exdb/mnist/>`_ is a set of hand-written
+    digits represented by grey-scale 28x28 images. Each pixel is scaled to
+    values in the interval ``[0, scale]``.
+
+    This function returns the training set and the test set of the official
+    MNIST dataset. If ``withlabel`` is True, each dataset consists of tuples of
+    images and labels, otherwise it only consists of images.
+
+    Args:
+        withlabel (bool): If True, it returns datasets with labels. In this
+            case, each example is a tuple of an image and a label. Otherwise,
+            the datasets only contain images.
+        ndim (int): Number of dimensions of each image. The shape of each image
+            is determined depending on ndim as follows:
+                - ``ndim == 1``: the shape is ``(784,)``
+                - ``ndim == 2``: the shape is ``(28, 28)``
+                - ``ndim == 3``: the shape is ``(1, 28, 28)``
+        scale (float): Pixel value scale. If it is 1 (default), pixels are
+            scaled to the interval ``[0, 1]``.
+
+    Returns:
+        A tuple of two datasets. If ``withlabel`` is True, both datasets are
+        :class:`~chainer.datasets.TupleDataset` instances. Othewrise, both
+        datasets are arrays of images.
+
+    """
+    train_raw = _retrieve_mnist_training()
+    train = _preprocess_mnist(train_raw, withlabel, ndim, scale)
+    test_raw = _retrieve_mnist_test()
+    test = _preprocess_mnist(test_raw, withlabel, ndim, scale)
+    return train, test
+
+
+def _preprocess_mnist(raw, withlabel, ndim, scale):
+    images = raw['x']
+    if ndim == 2:
+        images = images.reshape(-1, 28, 28)
+    elif ndim == 3:
+        images = images.reshape(-1, 1, 28, 28)
+    elif ndim != 1:
+        raise ValueError('invalid ndim for MNIST dataset')
+    images = images.astype(numpy.float32)
```

Comment:
Is it possible to change the dtype of images and labels different from float32 and int32, respectively?